### PR TITLE
Set Deny when the connector returns 404,403

### PR DESCRIPTION
**Pull request by shlomitk1**
Signed-off-by: Shlomit Koyfman <shlomitk@il.ibm.com>
Fixes https://github.com/fybrik/fybrik/issues/1711

### CODE REVIEW

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go [hunk 1]
```diff
@@ -15,6 +16,13 @@ import (
 	"fybrik.io/fybrik/pkg/model/datacatalog"
 )
 
+// ErrorMessages that are reported to the user
+const (
+	InvalidAssetID        string = "the asset does not exist"
+	AccessForbidden       string = "no permissions to access the data"
+	InvalidAssetDataStore string = "the asset data store is not supported"
```
Comment (revit13) [inline pkg/connectors/datacatalog/clients/datacatalog_openapi.go:23]: InvalidAssetDataStore -> DataStoreNotSupported

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go [hunk 2]
```diff
@@ -15,6 +16,13 @@ import (
 	"fybrik.io/fybrik/pkg/model/datacatalog"
 )
 
+// ErrorMessages that are reported to the user
+const (
+	InvalidAssetID        string = "the asset does not exist"
```
Comment (revit13) [inline pkg/connectors/datacatalog/clients/datacatalog_openapi.go:21]: why not use the specific causein the name?  InvalidAssetID -> AssetIDNotFound? invalid is general...

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go [hunk 3]
```diff
@@ -48,33 +56,45 @@ func NewOpenAPIDataCatalog(name, connectionURL string) DataCatalog {
...
+	defer httpResponse.Body.Close()
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("delete asset info from %s failed", m.name))
 	}
 	return resp, nil
 }
 
 func (m *openAPIDataCatalog) UpdateAsset(in *datacatalog.UpdateAssetRequest, creds string) (*datacatalog.UpdateAssetResponse, error) {
-	resp, _, err := m.client.DefaultApi.UpdateAsset(
+	resp, httpResponse, err := m.client.DefaultApi.UpdateAsset(
```
Comment (revit13) [inline pkg/connectors/datacatalog/clients/datacatalog_openapi.go:95]: httpResponse should be checked right?

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go [hunk 4]
```diff
@@ -48,33 +56,45 @@ func NewOpenAPIDataCatalog(name, connectionURL string) DataCatalog {
...
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("create asset info from %s failed", m.name))
 	}
 	return resp, nil
 }
 
 func (m *openAPIDataCatalog) DeleteAsset(in *datacatalog.DeleteAssetRequest, creds string) (*datacatalog.DeleteAssetResponse, error) {
-	resp, _, err := m.client.DefaultApi.DeleteAsset(context.Background()).XRequestDatacatalogCred(creds).DeleteAssetRequest(*in).Execute()
+	resp, httpResponse, err :=
+		m.client.DefaultApi.DeleteAsset(context.Background()).XRequestDatacatalogCred(creds).DeleteAssetRequest(*in).Execute()
```
Comment (revit13) [inline pkg/connectors/datacatalog/clients/datacatalog_openapi.go:86]: httpResponse should be checked right?

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go (hunk 2 above)
Comment (shlomitk1) [reply pkg/connectors/datacatalog/clients/datacatalog_openapi.go]: Changed.

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go (hunk 4 above)
Comment (shlomitk1) [reply pkg/connectors/datacatalog/clients/datacatalog_openapi.go:87]: Added a check for asset not found. All responses that are not successful return an error which is checked.

**Review by revit13 — APPROVED** [review]

File: pkg/connectors/datacatalog/clients/datacatalog_openapi.go [hunk 5]
```diff
@@ -48,33 +56,53 @@ func NewOpenAPIDataCatalog(name, connectionURL string) DataCatalog {
...
+	}
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("get asset info from %s failed", m.name))
 	}
 	return resp, nil
 }
 
 func (m *openAPIDataCatalog) CreateAsset(in *datacatalog.CreateAssetRequest, creds string) (*datacatalog.CreateAssetResponse, error) {
-	resp, _, err := m.client.DefaultApi.CreateAsset(context.Background()).
+	resp, httpResponse, err := m.client.DefaultApi.CreateAsset(context.Background()).
```
[the remark this answers is not in the document]
Comment (shlomitk1) [reply pkg/connectors/datacatalog/clients/datacatalog_openapi.go:75]: No, we still report an error.

**Pull request merged**